File: tck/java-tck/src/main/java/kalix/tck/model/eventsourcedentity/EventSourcedConfiguredEntity.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.eventsourcedentity;

import kalix.javasdk.eventsourcedentity.*;
import kalix.javasdk.eventsourcedentity.EventSourcedEntity;
import kalix.tck.model.eventsourcedentity.EventSourcedEntityApi.*;

public class EventSourcedConfiguredEntity extends AbstractEventSourcedConfiguredEntity {

  public EventSourcedConfiguredEntity(EventSourcedEntityContext context) {}

  @Override
  public Persisted emptyState() {
    return Persisted.getDefaultInstance();
  }

  public EventSourcedEntity.Effect<Response> call(Persisted currentState, Request request) {
    return effects().reply(Response.getDefaultInstance());
  }

  @Override
  public Persisted persisted(Persisted currentState, Persisted persisted) {
    return currentState;
  }
}

File: tck/scala-tck/src/main/scala/kalix/tck/model/eventsourcedentity/EventSourcedConfiguredEntity.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.eventsourcedentity

import kalix.scalasdk.eventsourcedentity.EventSourcedEntity

class EventSourcedConfiguredEntity extends AbstractEventSourcedConfiguredEntity {
  override def emptyState: Persisted = Persisted.defaultInstance

  override def call(currentState: Persisted, request: Request): EventSourcedEntity.Effect[Response] =
    effects.reply(Response.defaultInstance)

  override def persisted(currentState: Persisted, persisted: Persisted): Persisted =
    Persisted.defaultInstance
}
